nengo/builder/connection.py

```diff
@@ -53,6 +57,50 @@ def build_linear_system(model, conn, rng):
     return eval_points, activities, targets
 
 
+def build_decoders(model, conn, transform, rng):
+    encoders = model.params[conn.pre_obj].encoders
+    gain = model.params[conn.pre_obj].gain
+    bias = model.params[conn.pre_obj].bias
+
+    eval_points = get_eval_points(model, conn, rng)
+    targets = get_targets(model, conn, eval_points)
+
+    x = np.dot(eval_points, encoders.T / conn.pre_obj.radius)
+    E = None
+    if conn.solver.weights:
+        E = model.params[conn.post_obj].scaled_encoders.T
+        # account for transform
+        targets = np.dot(targets, transform.T)
+        transform = np.array(1., dtype=np.float64)
+
+    try:
+        decoders, solver_info = model.decoder_cache.wrap_solver(
+            solve_for_decoders)(
+                conn.solver, conn.pre_obj.neuron_type, gain, bias, x, targets,
+                rng=rng, E=E)
+    except ZeroActivityError:
+        raise ZeroActivityError(
+            "Building %s: 'activities' matrix is all zero for %s. "
+            "This is because no evaluation points fall in the firing "
+            "ranges of any neurons." % (conn, conn.pre_obj))
```

Comment:
Should this raise an exception, or should it just return a decoder that's all zeros?  It seems weird to me that adjusting neuron properties could suddenly cause an exception to be raised.